flexmeasures/api/v3_0/users.py

```diff
@@ -48,12 +48,15 @@ class UserAPI(FlaskView):
     @permission_required_for_context("read", ctx_arg_name="account")
     @as_json
     def index(self, account: Account, include_inactive: bool = False):
-        """API endpoint to list all users of an account.
+        """API endpoint to list all users.
+
 
         .. :quickref: User; Download user list
 
         This endpoint returns all accessible users.
         By default, only active users are returned.
+        The `account_id` query parameter can be used to filter the users of
```

Comment:
As I understand it, if no `account_id` is passed:
- The `AccountIdField` will default to the API user's own account.
- The  `permission_required_for_context` decorator will then do a read authorization check on the API user's own account (which should pass fine).
- `get_users` will return the all users under the API user's account.

Although still a work-in-progress, you suggested that these changes enable to fetch all accessible users in a single API call, rather than an API call per account. So, two preliminary questions:
1. when would `account is None` happen?
2. if this does now support listing users from all accessible accounts, I don't see where the right permissions check happens?

Reply:
Good point. 

> 1. when would account is None happen?

Apparently, other than tests, the `account` field is left undefined in the function `get_all_users` which is used in the `UserCrudUI` index page. If we want to keep the existing functionality, we can pass the `current_user.account.id` to the internal API call. 

> 2. if this does now support listing users from all accessible accounts, I don't see where the right permissions check happens? 

You are right, authorization wasn't being dealt properly in my change. The latest commit introduces some checks on the permissions of a user to an account.